cmd/gaia/init/gentx.go

```diff
@@ -86,8 +93,45 @@ following delegation and commission default parameters:
 					return err
 				}
 			}
-			// Run gaiad tx create-validator
+
+			// Set flags for creating gentx
 			prepareFlagsForTxCreateValidator(config, nodeID, ip, genDoc.ChainID, valPubKey)
+
+			// Fetch the amount of coins staked
+			amount := viper.GetString(cli.FlagAmount)
+			coins, err := sdk.ParseCoins(amount)
+			if err != nil {
+				return err
+			}
+
+			inGenesis := false
+			bondDenom := genesisState.StakeData.Params.BondDenom
+
+			// Check if the account is in genesis
+			for _, acc := range genesisState.Accounts {
```

Comment:
The contents of this for loop should probably be broken out into a seperate function, or this whole containing function restructured to reduce indentation